Semantics.md

```diff
@@ -862,18 +872,24 @@ thread wakes this one, this operator returns 2 ("timed-out").
 | 2 | "timed-out", not woken by another thread before timeout expired |
 
   * `i32.wait`: load i32 value, compare to expected, and wait for `i32.wake` at same address
-  * `i64.wait`: load i64 value, compare to expected, and wait for `i64.wake` at same address
 
 ### Wake
 
-The wake operators take two operands: an address operand, and a wake count. The
-operation will wake a maximum of "wake count" threads that are waiting on the
-same address. The operator returns the number of thread that were
-woken.
+The wake operator take two operands: an address operand, and a wake count as an
+`i32`. The operation will wake as many waiters as are waiting on the same
+effective address, up to the maximum as specified by `wake count`. The operator
+returns the number of waiters that were woken as an `i32`.
+
+| `wake count` value | Behavior |
+| ---- | ---- |
+| `wake count` < 0 | Wake all waiters |
+| `wake count` == 0 | Wake no waiters |
+| `wake count` > 0 | Wake up to `wake count` waiters |
 
-  * `i32.wake`: wake up to N threads waiting on the given address via `i32.wait`
-  * `i64.wake`: wake up to N threads waiting on the given address via `i64.wait`
+  * `i32.wake`: wake up `wake count` threads waiting on the given address via `i32.wait`
```

Comment:
I think you went a bit too far: if there aren't `wake count` waiters then you just wake all of them. The SAB algorithm does this IIUC. We could use `min(wake count , waiters)` or something?